pandas/core/base.py

```diff
@@ -990,13 +1009,94 @@ def factorize(self, sort=False, na_sentinel=-1):
         from pandas.core.algorithms import factorize
         return factorize(self, sort=sort, na_sentinel=na_sentinel)
 
-    def searchsorted(self, key, side='left'):
-        """ np.ndarray searchsorted compat """
+    _shared_docs['searchsorted'] = (
+        """Find indices where elements should be inserted to maintain order.
+
+        Find the indices into a sorted %(klass)s `self` such that, if the
+        corresponding elements in `v` were inserted before the indices, the
+        order of `self` would be preserved.
+
+        Parameters
+        ----------
+        v : array_like
+            Values to insert into `a`.
+        side : {'left', 'right'}, optional
+            If 'left', the index of the first suitable location found is given.
+            If 'right', return the last such index.  If there is no suitable
+            index, return either 0 or N (where N is the length of `a`).
+        sorter : 1-D array_like, optional
```

Comment:
I agree.  However, @shoyer said though that externalizing an argument such as `sorter` and then not using it was not really a good idea, hence why the `NotImplementedError` was used in `Index`.  Personally, I am not entirely sure which is the "proper" way to go, as either make sense.  Ideally, we would want to swallow it into `kwargs` arg so there is no externalization but that is not possible given that `numpy` calls the function internally with positional arguments.